src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -121,11 +124,92 @@ protected override void RemoveItem(int index)
 
             base.RemoveItem(index);
 
+            if (!_skipRaisingEvents)
```

Comment:
Can we hit race conditions here? Where we set this to true and then there is an individual RemoveItem call caused by a different thread rather than from the RemoveItemRange and miss an event? Or we don't guarantee thread safe operations here? cc: @stephentoub

Reply:
We don't need to be concerned with thread-safety here, as none of these operations promise that (and they're very much not thread-safe today).

Reply:
I agree, by design `ObservableCollection<T>` is not thread-safe, I don't think this is something we need to worry about.

If there is more changes we need on this, feel free to reopen